src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,133 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState, MouseEvent,
+} from 'react';
+import { Todo } from '../../types/Todo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as curentTodoActions } from '../../features/currentTodo';
+import { Filter } from '../../features/filter';
 
 export const TodoList: React.FC = () => {
+  const [filtredTodos, setFiltredTodos] = useState<Todo[]>([]);
+
+  const todos: Todo[] = useAppSelector(state => state.todos);
+  const filter: Filter = useAppSelector(state => state.filter);
+  const { query, status } = filter;
+
+  const dispatch = useAppDispatch();
+  const set = useCallback((curentTodo: Todo) => (
+    dispatch(curentTodoActions.setTodo(curentTodo))
+  ), []);
+
+  const setHandler = (event: MouseEvent<HTMLButtonElement>, todo: Todo) => {
+    event.preventDefault();
+    set(todo);
+  };
+
+  const isNoMathches = useMemo(() => Boolean(filtredTodos.length <= 0 && query),
+    [filter, filtredTodos]);
```

Comment:
Replace 'Boolean(filtredTodos.length <= 0 && query)' with 'filtredTodos.length === 0 && query' to simplify the expression.